R/mergeAttr.r

```diff
@@ -1,10 +1,21 @@
-mergeAttr <- function ( x, y, by = intersect(names(x), names(y)), by.x = by, by.y = by, all = FALSE, all.x = all, all.y = all, sort = TRUE, suffixes = c(".x",".y"), setAttr = TRUE, onlyVarValLabs = TRUE, homoClass = TRUE, unitName = "unit", xName = "x", yName = "y", verbose = c("match", "unique", "class", "dataframe")) {
+mergeAttr <- function ( x, y, by = intersect(names(x), names(y)), by.x = by, by.y = by, all = FALSE, all.x = all, all.y = all, sort = TRUE, suffixes = c(".x",".y"), setAttr = TRUE, onlyVarValLabs = TRUE, homoClass = TRUE, unitName = "unit", xName = "x", yName = "y", verbose = c("match", "unique", "class", "dataframe", "common")) {
      ### verbose setzen
-             verb  <- setVerbose(verbose, choices = c("match", "unique", "class", "dataframe"))
+             verb  <- setVerbose(verbose, choices = c("match", "unique", "class", "dataframe", "common"))
      ### das muessen data.frames sein
-             x <- makeDataFrame(x)
-             y <- makeDataFrame(y)
+             x     <- makeDataFrame(x)
+             y     <- makeDataFrame(y)
              byvars<- data.frame ( x=by.x, y=by.y, clx = sapply(x[,by.x,drop=FALSE], FUN = function(z) {paste(class(z),collapse="_")}), cly = sapply(y[,by.y,drop=FALSE], FUN = function(z) {paste(class(z),collapse="_")}), stringsAsFactors = FALSE)
+     ### schauen, ob zusaetzlich zu den by-Variablen noch weitere gemeinsame Variablen in den Datensaetzen existieren
+     ### die kriegen dann ja das suffix. Fall ja, message ausgeben
+             comm  <- intersect(colnames(x), colnames(y))
+             commby<- setdiff(comm, unique(unlist(byvars[,1:2])))

```

Comment:
LGTM
Nit: Would it be simpler (and easier to understand) to use `unique(c(by.x, by.y))` instead of the first two columns of `byvars`? It took me a bit to understand the structure of `byvars`.
Further nit: Maybe line breaks could be added to line 7?

Reply:
line breaks added. I would like to avoid `unique(c(by.x, by.y))` because that might go wrong if users specify `by` columns numerically.